pkg/reconciler/artifactbuild/artifactbuild.go

```diff
@@ -580,6 +591,29 @@ func (r *ReconcileArtifactBuild) handleRebuild(ctx context.Context, abr *v1alpha
 	abr.Status.SCMInfo = v1alpha1.SCMInfo{}
 	abr.Status.Message = ""
 	err := r.client.Status().Update(ctx, abr)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+
+	//now delete old pipelines
```

Comment:
just for my education @stuartwdouglas - is deleting old pipelineruns actually part of private source code repo support, or is this just an optimization you added while you were in here?

If the latter, I think I see why that is OK, since we are setting our state back to new in line 590 as part of presumably a rebuild, right?